spec/take_away_ui_spec.rb

```diff
@@ -0,0 +1,215 @@
+        expect(menu.cuisine_choice).to eq dummy_cuisine_klass
+      end
+
+      it 'has an uninstantiated Order object' do
+        expect(menu.order).to eq dummy_order_klass
+      end
+
+      it 'has an uninstantiated Basket object' do
+        expect(menu.basket).to eq dummy_basket_klass
+      end
+
+      it 'has a selection array' do
+        expect(menu.selection).to be_a Array
+      end
+
+      it 'has an empty selection array as default' do
+        expect(menu.selection).to be_empty
+      end
+    end
+
+  context "Choosing and displaying menus" do
+
+    describe '#menu_choice' do
+
+      it 'will raise error "We only serve lunch and dinner" if another meal is asked for' do
+        expect{menu.menu_choice(:breakfast)}.to raise_error("We only serve lunch and dinner")
+      end
+
+      it 'will raise error "Please choose starter, main or dessert" if another meal is asked for' do
+        expect{menu.menu_choice(:dinner, :snack)}.to raise_error("Please choose starter, main, dessert or 'all'")
+      end
+
+      it 'returns a hash of the dishes selected with menu_choice' do
+        expect(menu.menu_choice).to be_a Hash
+      end
+
+      it 'selects a menu according to meal and course' do
+        expect(menu.menu_choice).to eq default_menu_dinner_mains
+      end
+
+
+    describe '#display' do
+
+      before do
+        allow(italian_menu).to receive(:each)
+      end
+
+      it 'displays the dishes and prices of the ItalianMenu, defaulted to Dinner/Mains' do
+        expect(menu.display).to eq default_menu_dinner_mains
+      end
+    end
+  end
+end
+
+
+
+  context 'Submiting orders by text or selecting and adding to basket individually' do
+        let(:test_order) {"2 carbonara, 1 tiramisu and I dunno"}
+        let(:selection1) {["Cake",3]}
+        let(:selection2) {["Tiramisu", 1]}
+
+
+        before do
+          allow(STDOUT).to receive(:p).with("What would you like to eat? Eg '2 Carbonara and a Tiramisu'")
+          allow(STDIN).to receive(:gets).and_return(test_order)
+          allow(dummy_order_klass).to receive(:new).with(test_order, default_menu_dinner_mains).and_return(dummy_order)
+          allow(dummy_basket_klass).to receive(:new).with([selection1], italian_menu).and_return(dummy_basket)
+        end
+
+        describe '#take_order' do
+
+          before do
+            allow(dummy_order).to receive(:add_to_basket)
+          end
+
+
+          it 'instantiates a new order' do
+            expect(dummy_order_klass).to receive(:new).and_return(dummy_order)
+            menu.take_order
+          end
+
+          it 'adds the order to basket' do
+            expect(dummy_order).to receive(:add_to_basket)
+            menu.take_order
+          end
+        end
+
+      describe '#select_dish' do
+          it {is_expected.to respond_to(:select_dish).with(2).arguments}
+
+
+          before do
+            allow(dummy_basket).to receive(:selection).and_return([[selection1]])
```

Comment:
Metrics/LineLength: Line is too long. [81/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)